language-service/src/services/yamlCompletion.ts

```diff
@@ -85,8 +87,17 @@ export class YAMLCompletion {
 			nodeEndPos.character += 1;
 			overwriteRange = Range.create(nodeStartPos, nodeEndPos);
 		}else if (node && (node.type === 'string' || node.type === 'number' || node.type === 'boolean')) {
-			//console.log('type = string | nuber | boolean');
-			overwriteRange = Range.create(document.positionAt(node.start), document.positionAt(node.end));
+            //console.log('type = string | nuber | boolean');
+            const startPosition = document.positionAt(node.start);
+            let endPosition = document.positionAt(node.end);
+
+            // when start and end positions are not on the same line and node is a temporary holder then
+            // we must have misplaced the end of the range one line below the start of the range
+            // because of the mismatch in line length between the temporary currentDoc with holder and actual document.
+            if (startPosition.line + 1 === endPosition.line && endPosition.character === 0 && node.location === nodeHolder) {
```

Comment:
Is the `endPosition.character === 0` check necessary?  It turns out to be true because of the `h` but doesn't seem necessary here.

Reply:
I want to minimize the possibilities of this incorrectly fixing something that is intentional. For example, it is valid to have your range as multi-line:
```
- h:
  <ctrl+space here>
  foo: bar
```
Since nodeHolder is one character the range will spill maximum to 2 lines because each line will have at least one character '\n'. That means the character position of the second line can maximum will be 0. If it is not zero there are chances that this is an intentional multi-line suggestion.

Reply:
I don't think multi-line suggestions are currently supported by our implementation, mostly because of this holder hack. But still I think it is better to have more specific checks.